drop-transfer/src/check.rs

```diff
@@ -7,56 +7,107 @@ use tokio_util::sync::CancellationToken;
 use crate::{protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
 
 pub(crate) fn spawn(
+    refresh_trigger: tokio::sync::watch::Receiver<()>,
     state: Arc<State>,
     xfer: Arc<IncomingTransfer>,
     logger: Logger,
     guard: AliveGuard,
     stop: CancellationToken,
 ) -> tokio::task::JoinHandle<()> {
     let id = xfer.id();
-    let job = run(state, xfer, logger.clone());
+    let refresh_trigger = refresh_trigger;
 
     tokio::spawn(async move {
         let _guard = guard;
+        loop {
+            let job = run(state.clone(), xfer.clone(), logger.clone());
 
-        tokio::select! {
-            biased;
+            tokio::select! {
+                biased;
 
-            _ = stop.cancelled() => {
-                debug!(logger, "Check job stop for transfer: {id}");
-            },
-            _ = job => (),
+                _ = stop.cancelled() => {
+                    debug!(logger, "Check job stop: {id}");
+                    break;
+                },
+                res = job => {
+                    debug!(logger, "Check job done: {id} -> {:?}", res);
+                    match res {
+                        ControlFlow::Break(_) => {
+                            break;
+                        }
+                        ControlFlow::Continue(_) => {
+                            debug!(logger, "Check job continue: {id}");
+
+                            match refresh_trigger.has_changed() {
+                                Ok(true) => debug!(logger, "Check job refresh: {id}"),
+                                Ok(false) => {
+                                    debug!(logger, "Check job no refresh: {id}");
+                                    break;
```

Comment:
By breaking the loop here you're actually canceling the task, which should not happen here.

Reply:
I want to break the task if I have already finished it and there was no API signal to retry again and I want to spin the loop again in case I received API call while the task was running.

Reply:
Are you sure? I imagine you just want to wait for the channel trigger and do it again, not break the task